### PR TITLE
Use ParserK explicitly

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fixes:https://github.com/composewell/streamly/issues/2257

File: core/src/Streamly/Internal/Data/Parser/ParserK/Type.hs [hunk 1]
```diff
@@ -143,7 +142,7 @@ newtype Parser a m b = MkParser
         -> m (Step a m r)
     }
 
-type ParserK = Parser
+--type ParserK = Parser
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Parser/ParserK/Type.hs:145]: Can remove this.

File: core/src/Streamly/Internal/Data/Stream/StreamK.hs [hunk 2]
```diff
@@ -1283,7 +1283,7 @@ parserDone (ParserK.Failure n e) _ _ = pure $ ParserK.Error n e
 {-# INLINE_NORMAL chunkParseBreak #-}
 chunkParseBreak
     :: (Monad m, Unbox a)
-    => ParserK.Parser a m b
+    => ParserK.ParserK a m b
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Stream/StreamK.hs:1286]: Use the type unqualified.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by adithyaov — APPROVED** [review]
Ah, I expected more changes for parser. Nice. There were only a few.

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**